Source/NSStoryboardSegue.m

```diff
@@ -106,11 +115,29 @@ - (void) perform
   // Perform segue based on it's kind...
   if ([_kind isEqualToString: @"relationship"])
```

Comment:
You really need to split up this huge method in different levels of dispatching.

Reply:
Yes, I had made an observation... this actually needs to be divided into separate subclasses, I believe.